### PR TITLE
Details Page Shared projects tab

### PR DESCRIPTION
**Pull request by joshsadam**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This pull request adds a 'Shared projects' tab to the group show page, allowing members of a Group to see which projects have been shared with it easily.

Included in this PR:

- Empty State view component
  - Updated empty state for the 'Subgroups and projects' tab on the group show page
  - Updated empty state for Sample attachments
- Shared projects table
  - Uses shared component with the dashboard projects for consistent look and feel 

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

**Shared projects tab with shared projects**
image

**Shared projects tab with empty state**
image

**Updated empty state on "Subgroups and projects" tab**
image

**Updated empty state on Sample files page**
image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: test/system/groups/show_test.rb
Comment (ericenns) [inline test/system/groups/show_test.rb:1]: **nitpick (non-blocking):** Is there a specific reason that this wasn't included in the normal groups system test. i.e. `test/system/groups_test.rb`

**Review by ericenns — CHANGES_REQUESTED** [review]
Few comments, otherwise functioning as expected.

File: app/controllers/groups/subgroups_controller.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Groups
+  # Controller actions for group subgroups and projects
+  class SubgroupsController < ApplicationController
+    before_action :group, only: %i[index]
+
+    def index
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply app/controllers/groups/subgroups_controller.rb:8]: chore: Updated in fdb782a911403d6a40b3f18fad52bc4ccf37e534

**Review by ericenns — APPROVED** [review]
Thanks for making the changes, I asked @deepsidhu85 to give a second review

**Review by deepsidhu85 — APPROVED** [review]
This looks good to me and is working as expected!

**Pull request merged**